assets/model_monitoring/components/src/model_data_collector_preprocessor/trace_aggregator.py

```diff
@@ -44,6 +56,10 @@ def aggregate_spans_into_traces(
 
     print("Processing spans into aggregated traces...")
 
+    # for PromptFlow we need to replace the trace_id values with request_id in order to handle edge cases
+    # where PF has same trace_id but multiple LLM requests
+    enlarged_span_logs = enlarged_span_logs.rdd.flatMap(_replace_trace_with_request_id).toDF(enlarged_span_logs.schema)
```

Comment:
Why use flatMap instead of map? This is a 1:1 mapping.

Reply:
No reason. We should use map() instead of flatMap